### ISSUE TITLE
[BUG] Masked Elements masking elements in fore front

### ISSUE THREAD

**Issue by gselsidi** [microsoft/playwright#19002]
Not sure if this is a feature request or bug.

Masked elements block out all other elements, even if the other elements are in the forefront, and the masked elements are in the background.

Is there a way to reverse this? Like If I'm masking an element and take a full screen snapshot with a pop out, the image shows the masked elements blocking out the element in the forefront...

Or is this a react issue?

**Comment by dgozman** [comment]
@gselsidi This behavior is intended, because it was deemed more useful in many scenarios. If you'd like to mask elements at their natural z-order, that's a fine feature request. I'll leave the issue for further prioritization.

**Comment by zakhenry** [comment]
I'd like to add a voice to this request. Often I would like to screenshot a modal or a bottom sheet, but the background is not something that is relevant (it is dimmed/blurred) but it is annoying to review changes that appear only in the background of these modals.

An obvious solution would be to screenshot just the modal, however that isn't actually that useful either, as it is important to our application that at certain breakpoints the bottom sheet or modal does not exceed a certain size obscuring some other UI element.

In short my preference would be for the mask to appear with the same z-index as the masked element, but immediately following it in the dom so that it obscures only the element that is masked and no other.

I'm curious of the scenarios you mention where the current behaviour would be preferred?

**Comment by janett-baresel** [comment]
Created a feature request with this exact same idea https://github.com/microsoft/playwright/issues/28010

**Comment by gselsidi** [comment]
should probably just add an option to the screenshot methods for z index, such as a true false boolean of ignore z index or not.

**Comment by dgozman** [comment]
You can now use the [`stylePath` option](https://playwright.dev/docs/test-snapshots#stylepath) for screenshots, and apply any styles you'd like during screenshot. For example, hiding some elements, making them entirely transparent or replacing them with a solid color placeholder.
